net/udp-broadcast-relay-redux-openwrt/files/udp-broadcast-relay-redux.init

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2006-2011 OpenWrt.org
+
+START=90
+STOP=10
+
+USE_PROCD=1
+PROG=/usr/sbin/udp-broadcast-relay-redux
+NAME=udp-broadcast-relay-redux
+PIDCOUNT=0
+
+validate_section_udp_broadcast_relay_redux()
+{
+    uci_validate_section udp_broadcast_relay_redux udp_broadcast_relay_redux "${1}" \
+	'id:uinteger' \
+	'port:port' \
+	'network:list(string)' \
+	'src_override:ip4addr' \
+	'dest_override:ip4addr'
+
+    [ -z "$id" ] && return 1
+
+    [ -z "$network" ] && return 1
+
+    [ -z "$port" ] && return 1
+
+    return 0
+}
+
+udp_broadcast_relay_redux_instance() {
+    local net network ifname id port src_override dest_override
+
+    validate_section_udp_broadcast_relay_redux "${1}" || {
+	echo "Validation failed"
+	return 1
+    }
+
+    PIDCOUNT="$((  ${PIDCOUNT} + 1))"
+
+    procd_open_instance
+    procd_set_param command "$PROG" "--id" "${id}" "--port" "${port}"
+
+    for net in $network; do
+	network_get_device ifname "$net"
+	if [ -z "$ifname" ]; then
+	    network_get_physdev ifname "$net"
+	fi
+	if [ -n "$ifname" ]; then
+	    procd_append_param command "--dev" "$ifname"
+	    procd_append_param netdev "$ifname"
+	fi
+    done
+
+    if [ ! -z "$src_override" ] ; then
+        procd_append_param command "-s" "$src_override"
+    fi
+
+    if [ ! -z "$dest_override" ] ; then
+        procd_append_param command "-t" "$dest_override"
+    fi
+
+    procd_add_jail ubr-${PIDCOUNT}
+    procd_close_instance
+}
+
+start_service() {
+    . /lib/functions.sh
+    . /lib/functions/network.sh
+
+    config_load udp_broadcast_relay_redux
+    config_foreach udp_broadcast_relay_redux_instance udp_broadcast_relay_redux
+}
+
+service_triggers() {
+    procd_add_reload_trigger "udp-broadcast-relay-redux"
```

Comment:
```suggestion
    procd_add_reload_trigger "udp_broadcast_relay_redux"
```
This should match the config name.

Reply:
> Why do you comment out all of the config?

I wondered about that as well (to comment it out or not).  To me, it didn't seem like my example config was ever likely to be desired by anyone without them first tweaking it for their use case (specific computer game or other use case).  Hence why I felt commenting it out was better.

But, if you think it's more sane to leave it un-commented, I'm also fine with that!  Admittedly I'm kinda unsure about all this stuff so I'll definitely defer to you!

Reply:
> Hence why I felt commenting it out was better.

Oh I see, and thanks for explanation.
But please at least keep `config udp_broadcast_relay_redux` uncommented.

Reply:
Ok, is that expected to work (config section that's empty)?

Reply:
It's okay.

Reply:
Hmm, I just tested it with the following config:

```
config udp_broadcast_relay_redux
#       option id 1
#       option port 47624
#       list network lan
#       list network vpnsrv
#       list network vpnsuperior
#       option dest_override 255.255.255.255
```

With this config:

```
root@wrt1200ac-Ocqueoc:~# /etc/init.d/udp-broadcast-relay-redux start
Validation failed
```

Whereas if it's entirely commented out 

```
/etc/init.d/udp-broadcast-relay-redux start
```
Does nothing.  Which is preferred?

Reply:
In my option, doing nothing is better :-)